### PR TITLE
Handle process.env being undefined in debug func

### PR DESCRIPTION
**Pull request by jkarthic**
- [x] I understand that this repository is auto-generated and my pull request may not be merged

## Changes being requested
Please handle the case where `process.env` is undefined.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-  "name": "openai",
+  "name": "@get-wrecked/openai",
   "version": "4.29.0",
   "description": "The official TypeScript library for the OpenAI API",
   "author": "OpenAI <support@openai.com>",
   "types": "dist/index.d.ts",
   "main": "dist/index.js",
   "type": "commonjs",
-  "repository": "github:openai/openai-node",
+  "repository": "github:get-wrecked/openai-node",
```
Comment (rattrayalex) [inline package.json:9]: These changes are, I hope, unintentionally left in from development

File: src/core.ts [hunk 2]
```diff
@@ -1075,7 +1075,7 @@ function applyHeadersMut(targetHeaders: Headers, newHeaders: Headers): void {
 }
 
 export function debug(action: string, ...args: any[]) {
-  if (typeof process !== 'undefined' && process.env['DEBUG'] === 'true') {
+  if (process?.env?.['DEBUG'] === 'true') {
```
Comment (rattrayalex) [inline src/core.ts:1078]:
```suggestion
  if (typeof process !== 'undefined' && process?.env?.['DEBUG'] === 'true') {
```
we still need the typeof check for environments where `process` is not defined at all.

**Review by rattrayalex — COMMENTED** [review]
Thanks for the contribution! Few changes requested.

File: package.json (hunk 1 above)
Comment (jkarthic) [reply package.json]: yes. Sorry about that. All fixed now. 
Thanks for the quick response.

**Review by rattrayalex — APPROVED** [review]

**Comment by rattrayalex** [comment]
Thank you @jkarthic !

**Comment by dackerman** [comment]
Thanks @jkarthic , we merged an update to our codegen to reflect your suggestion. It should go out in the next release!

**Comment by jkarthic** [comment]
Thank you very much @rattrayalex @dackerman

**Pull request closed without merging**